### DISCUSSION TITLE
Why result of mpc is different to the real robot pose?

### DISCUSSION THREAD

**Discussion by shiweizheng** [NVlabs/curobo#216] [Software Q&A]
I configured my own fanuc robot and run mpc_example.py. Then I got a reslut which shows MPC is success. But the result of MPC is different with my goal Pose.For example, I set start joint state is [ 0.0000,  0.0000,  0.2620,  0.0000, -0.2620,  0.0000], and goal joint state is [0.5000, 0.5000, 0.7620, 0.5000, 0.2380, 0.5000]. Finally I get a result and the last of trajotary is [0.4964, 0.5026, 0.7873, 0.4158, 0.2460, 0.5890], which is not same as the goal state.
![image](https://github.com/NVlabs/curobo/assets/19202332/3b8d4263-dc65-4a28-9490-fd7a42170c83)

Another strange thing, the goal state is set [ 0.0000,  0.0000,  0.2620,  0.0000, -0.2620,  0.0000], then the Pose calculated is  "ee_pos_seq=tensor([[ 0.7357, -0.5725,  0.9365]], device='cuda:0'), ee_quat_seq=tensor([[0.5191, 0.6769, 0.1262, 0.5064]]", which is not equal to real robot pose. That seems the result of IK is wrong?
![image](https://github.com/NVlabs/curobo/assets/19202332/50212637-eb3a-4f18-8876-3eb97b66c0ec)
Is this correct? And which I did wrong?Thank a lot!!

**Comment by balakumar-s** [comment]
What robot are you using?

**Reply by shiweizheng** [reply]
Thanks. Fanuc crx-10ia. The URDF file is view by other project, and it seems correct.
![20240412-093704](https://github.com/NVlabs/curobo/assets/19202332/f0e07ec9-4527-4d4f-8843-7449281684aa)
And the URDF file is here.
[fanuc_crx_urdf_meshes.zip](https://github.com/NVlabs/curobo/files/14953048/fanuc_crx_urdf_meshes.zip)
I was stuck several days. thanks a lot

**Reply by balakumar-s** [reply]
Did you create a cuRobo configuration file for the robot? If so can you share that as well to help us debug.

**Reply by shiweizheng** [reply]
Sorry so late! Thank you
Here is my configuration file about fanuc robot. 
[configuration.zip](https://github.com/user-attachments/files/15801045/configuration.zip)

**But**, i think maybe there is another my mistake.I run ik_example.py using official ur5e configuration, and the IK result is also wrong.
 I set target positon is [-0.81726, -0.2329, 0.0628] and quaternion is [0.7071068, 0.7071068, 0.0,  0.0]. The supposed IK result(degree) is joints [0,0,0,0,0,0], but the result(degree) is [167.3806915283203, -8.563478469848633, 13.385866165161133, -54.911800384521484, -37.77490997314453, 39.57598114013672]. I'm very confused.
![p1](https://github.com/NVlabs/curobo/assets/19202332/ede7670e-ddbc-4408-8a2a-3304fd1793b9)
![lxiZYSltLe](https://github.com/NVlabs/curobo/assets/19202332/9dc0d81e-d82a-449a-b388-88d312e1bd37)


And i also test my fanuc configuraton. result is also wrong. joint 5 is supposed -90 degree，but  it is 90 degree.
![boXfm5frmm](https://github.com/NVlabs/curobo/assets/19202332/41c84cee-880f-4033-989d-5855fabfe6db)
![image](https://github.com/NVlabs/curobo/assets/19202332/d4f5c3a5-a722-4905-b90c-0f4ca24359f4)

I think maybe that wrong IK result is the reason about wrong mpc.I am new to robot. Special thanks to your help.
